draft-ietf-tls-tls13.md

```diff
@@ -3524,12 +3524,11 @@ that:
   types. That is, if a handshake message is split over two or more
   records, there MUST NOT be any other records between them.
 
-- Handshake messages MUST NOT span key changes. Because
-  the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate messages
-  can arrive immediately prior to a key change,
-  upon receiving these messages a receiver MUST verify that the end
-  of these messages aligns with a record boundary; if not, then it MUST
-  terminate the connection with an "unexpected_message" alert.
+- Handshake messages MUST NOT span key changes. Implementations MUST verify that
+  all messages immediately preceding a key change align with a record boundary.
+  Because the ClientHello, EndOfEarlyData, ServerHello, Finished, and KeyUpdate
+  messages can immediately precede a key change, implementations MUST send these
+  messages in alignment with a record boundary.
```

Comment:
It's not really plausible for a client to fail to end a ClientHello on a record boundary because it completes a flight. I suppose you could predict that you're going to get an HRR and send the first few bytes of CH2 header early, but that's kind of nuts. I'm not particularly offended by trying to enumerate sender obligations by handshake messages if that's the simplest phrasing, but there isn't much point beyond it.

I agree that a MUST-level requirement on the receiver to check them should be tight and exclude the HRR case.